datastore_mongo.go

```diff
@@ -85,6 +85,23 @@ func (db *DataStoreMongo) GetDevice(id DeviceID) (*Device, error) {
 	return &res, nil
 }
 
+func (db *DataStoreMongo) DeleteDevice(id DeviceID) error {
+	s := db.session.Copy()
+	defer s.Close()
+
+	filter := bson.M{"id": id}
```

Comment:
Run ensure index on application startup (then initing db object for example) with building index in foreground to make sure we start application with existing index.

https://github.com/mendersoftware/deployments/blob/master/resources/images/mongo/images.go#L64